src/utilities/use-is-mounted-ref.ts

```diff
@@ -0,0 +1,19 @@
+import {useRef, useEffect, MutableRefObject} from 'react';
+
+/**
+ * Returns a MutatableRefObject containing a boolean value that
+ * represents a components mounted status.
+ * @returns MutableRefObject<boolean> The mounted status
+ */
+export function useIsMountedRef(): MutableRefObject<boolean> {
+  const isMounted = useRef(false);
```

Comment:
Should this be true so the hook returns true on a component's first render?

Reply:
The component isn't mounted during the first render, however, it'll be mounted when the effects run? afaik

Reply:
From http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/ and https://github.com/donavon/hook-flow it looks like you're right - I guess those "­React updates ­D­O­M " phases are when something goes from not being mounted to is mounted so in that render before first mount a component is not mounted. I was thrown because the equivalent hook in quilt sets this to true, but it seems like that's a bug on their side: https://github.com/Shopify/quilt/blob/master/packages/react-hooks/src/hooks/mounted-ref.ts

Reply:
I had mentioned this to @michenly and it sounds like this is the intention. From slack: `I think you should think of this line more like a constructor of the class and the value will remain true till componentWillUnmount... it is not equal to componentDidMount... I wonder if it should be renamed to useUnMountedRef`,

Reply:
Typically where we are using `useMountedRef` from quilt are for checking if something had become un-mounted, so even thought the implementation is not exactly accurate it had been working so far.

I would like to fix it as a bug, but wonder if we would run into any situation where going from false -> true - > false will introduce a flash in UI. So this might need to be a breaking change with a more clear new name to avoid confusion.